daml-lf/archive/BUILD.bazel

```diff
@@ -17,54 +17,88 @@ load(
 load("//bazel_tools:pom_file.bzl", "pom_file")
 load("@os_info//:os_info.bzl", "is_windows")
 load("//bazel_tools:javadoc_library.bzl", "javadoc_library")
+load(":archive.bzl", "mangle", "mangle_for_java")
+load("//bazel_tools:java.bzl", "da_java_proto_library")
 
 LF_MAJOR_VERSIONS = [
     "0",
     "1",
 ]
 
-filegroup(
-    name = "daml-lf-archive-srcs",
-    srcs = ["da/daml_lf.proto"] + ["da/daml_lf_%s.proto" % v for v in LF_MAJOR_VERSIONS],
-    visibility = ["//visibility:public"],
-)
-
-exports_files(
-    ["da/daml_lf_%s.proto" % v for v in LF_MAJOR_VERSIONS],
-)
+LF_VERSIONS = [
+    "daml-lf",
+    "daml-lf-1.6",
+]
 
-proto_library(
-    name = "daml_lf_proto",
-    srcs = [":daml-lf-archive-srcs"],
-    proto_source_root = "daml-lf/archive",
-    visibility = ["//:__subpackages__"],
-)
+[
+    [
+        filegroup(
+            name = "%s_archive_proto_srcs" % mangled_version,
+            srcs = glob(["src/main/protobuf/com/digitalasset/%s/*.proto" % mangle_for_java(mangled_version)]),
+        ),
+        proto_library(
+            name = "%s_archive_proto" % mangled_version,
+            srcs = ["%s_archive_proto_srcs" % mangled_version],
+            strip_import_prefix = "src/main/protobuf/",
+            visibility = ["//visibility:public"],
+            deps = ["@com_google_protobuf//:empty_proto"],
+        ),
+        da_java_proto_library(
```

Comment:
Also add a bazel target for the javadoc and sources.

Reply:
done